### ISSUE TITLE
The weak_ref_create should be able to take in an Id.Instance and be able to create an weak_ref to it

### ISSUE THREAD

**Issue by DiasFranciscoA** [YoYoGames/GameMaker-Bugs#6778]
### Description

right now the only way to create a weak ref to an instance ID is by doing the following:

```
// lets say I have the instance id stored here
// Note that if method IS bound to an instance this will return the id not the self of the instance
_id = method_get_self(_method_bound_to_the_instance);

var _self = undefined;
with (_id) {
   _self = self;
}

// Now _self is the self of the struct or instance
weak_ref_create(_self); // this will work

```

It would be good to avoid this by changing `weak_ref_create` to allow you to pass an Id.Instance also.

```
_self_or_id = method_get_self(_method_bound_to_the_instance);
weak_ref_create(_self_or_id ); // this would work

```

[allow_weakref_create_on_instances.zip](https://github.com/user-attachments/files/16310763/allow_weakref_create_on_instances.zip)

### Expected Change

Allow weak_ref_create to also be passed Id.Instance

### Steps To Reproduce

1. Start GameMaker
2. 
3. See the issue

### How reliably can you recreate this issue using your steps above?

Always

### Which version of GameMaker are you reporting this issue for?

2024.800 (Betas)

### Which platform(s) are you seeing the problem on?

Windows

### Contact Us Package Attached?

- [X] I have attached my Contact Us Package

### Sample Project Added?

- [X] I have included a small sample project

**Comment by rwkay** [comment]
Fixed in 2024.8

**Comment by YYBartT** [comment]
Added to the manual page that `weak_ref_create()` can take an instance. Removed method as a parameter type (creates a weak ref to the method itself, rather than to the struct).

**Comment by scott-dunbar** [comment]
Verified fixed in Beta IDE v2024.800.0.613 Beta Runtime v2024.800.0.636, confirmed additions to manual are published
